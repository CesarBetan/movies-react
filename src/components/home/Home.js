import React from 'react';
import ShowsSlider from '../showsSlider/ShowsSlider';
import { buildUrl } from '../../utils/api';

const Home = () => {
  return (
    <div className="home-wrapper">
      <ShowsSlider
        titleShows={'Popular'}
        movieApiUrl={buildUrl('popular')}
        endpoint={'popular'}
        viewOn={true}
      />
      <ShowsSlider
        titleShows={'Top Rated'}
        movieApiUrl={buildUrl('top_rated')}
        endpoint={'top_rated'}
        viewOn={true}
      />
      <ShowsSlider
        titleShows={'Now Playing'}
        movieApiUrl={buildUrl('now_playing')}
        endpoint={'now_playing'}
        viewOn={true}
      />
    </div>
  );
};

export default Home;
