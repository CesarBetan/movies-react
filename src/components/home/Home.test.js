import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import Home from './Home';

describe('HomeTest', () => {
  test('component should render', () => {
    const history = createMemoryHistory();
    const container = render(
      <Router history={history}>
        <Home />
      </Router>,
    );
    expect(container).toMatchSnapshot();
  });
});
