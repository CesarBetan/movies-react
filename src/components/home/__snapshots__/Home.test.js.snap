// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeTest component should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home-wrapper"
      >
        <div
          class="shows-slider"
        >
          <h2
            class="shows-title"
          >
            Popular
          </h2>
          <div
            class="shows-slider-content"
          >
            <span>
              loading...
            </span>
            <div
              class="viewall"
            >
              <a
                href="/popular/view_all"
              >
                View All
              </a>
            </div>
          </div>
        </div>
        <div
          class="shows-slider"
        >
          <h2
            class="shows-title"
          >
            Top Rated
          </h2>
          <div
            class="shows-slider-content"
          >
            <span>
              loading...
            </span>
            <div
              class="viewall"
            >
              <a
                href="/top_rated/view_all"
              >
                View All
              </a>
            </div>
          </div>
        </div>
        <div
          class="shows-slider"
        >
          <h2
            class="shows-title"
          >
            Now Playing
          </h2>
          <div
            class="shows-slider-content"
          >
            <span>
              loading...
            </span>
            <div
              class="viewall"
            >
              <a
                href="/now_playing/view_all"
              >
                View All
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home-wrapper"
    >
      <div
        class="shows-slider"
      >
        <h2
          class="shows-title"
        >
          Popular
        </h2>
        <div
          class="shows-slider-content"
        >
          <span>
            loading...
          </span>
          <div
            class="viewall"
          >
            <a
              href="/popular/view_all"
            >
              View All
            </a>
          </div>
        </div>
      </div>
      <div
        class="shows-slider"
      >
        <h2
          class="shows-title"
        >
          Top Rated
        </h2>
        <div
          class="shows-slider-content"
        >
          <span>
            loading...
          </span>
          <div
            class="viewall"
          >
            <a
              href="/top_rated/view_all"
            >
              View All
            </a>
          </div>
        </div>
      </div>
      <div
        class="shows-slider"
      >
        <h2
          class="shows-title"
        >
          Now Playing
        </h2>
        <div
          class="shows-slider-content"
        >
          <span>
            loading...
          </span>
          <div
            class="viewall"
          >
            <a
              href="/now_playing/view_all"
            >
              View All
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
